alpha/lib/model/category.php

```diff
@@ -1918,7 +1918,7 @@ private function getDescendantCategoriesIds()
 	 */
 	public function getElasticIndexName()
 	{
-		return ElasticIndexMap::ELASTIC_CATEGORY_INDEX;
+		return kBaseESearch::getElasticIndexNamePerPartner(ElasticIndexMap::ELASTIC_CATEGORY_INDEX, $this->getPartnerId());
```

Comment:
kBaseESearch is in esearch plugin, maybe we should move esearch plugin to be core as it becomes more and more mandatory to our system, otherwise, we might need to change all the calls to kBaseESearch to be more compatible with how we use plugins

Reply:
Move the entire logic into the plugin (this was a good point)